sqlalchemy_hana/dialect.py

```diff
@@ -230,16 +230,23 @@ def set_isolation_level(self, connection, level):
                     cursor.execute("SET TRANSACTION ISOLATION LEVEL %s" % level)
 
     def get_isolation_level(self, connection):
-        with connection.cursor() as cursor:
-            cursor.execute("SELECT CURRENT_TRANSACTION_ISOLATION_LEVEL FROM DUMMY")
-            result = cursor.fetchone()
+        # In this case the 'with' statement on connection.cursor() as cursor is not 
+        # working for pyHDB.
+        cursor = connection.cursor()
+        cursor.execute("SELECT CURRENT_TRANSACTION_ISOLATION_LEVEL FROM DUMMY")
+        result = cursor.fetchone()
         return result[0]
 
     def _get_server_version_info(self, connection):
         pass
 
     def _get_default_schema_name(self, connection):
-        return self.normalize_name(connection.engine.url.username.upper())
+        # In this case, the SQLAlchemy Connection object is not yet "ready".
```

Comment:
Yes PyHDB's cursor object doesn't implement `__enter__` and `__exit__` well but now the cursor will be stay open. I recommend to reuse the pattern from that [line](https://github.com/SAP/sqlalchemy-hana/blob/4989e6e7d9bea51b0843e49257624f2fc9ca05b1/sqlalchemy_hana/dialect.py#L235) and use the `contextlib. closing` helper function.